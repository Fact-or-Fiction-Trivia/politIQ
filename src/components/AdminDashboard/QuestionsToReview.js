import React from 'react';
import { db } from '../../firebase';
import { Link } from 'react-router-dom';

import { REVIEW } from '../../constants/routes';

import './dashboard.css';

const QuestionsToReview = () => {
  return (
    <Link to={REVIEW} style={{ textDecoration: 'none' }}>
      <div className="reviewWidget">
        <div className="counterDisplay">{0}</div>
        <h1>Questions to be reviewed</h1>
      </div>
    </Link>
  )
}

export default QuestionsToReview;
