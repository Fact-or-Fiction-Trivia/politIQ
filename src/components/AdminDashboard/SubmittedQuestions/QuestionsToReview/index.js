import React, { Component } from 'react';
import { db } from '../../../../firebase';
import { Link } from 'react-router-dom';

import { REVIEW } from '../../../../constants/routes';

import '../../dashboard.css';

class QuestionsToReview extends Component {
  constructor(props) {
    super(props);
    this.state = {
      num: 0
    }
  }

  componentDidMount = () => {
    this.numToReview()
  }

  componentWillUnmount = () => {
    this.setState({
      undefined
    })
  }

  numToReview = async () => {
    // let num;
    await db.getOneQuestion()
      .then(response => {
        if (response.val() === null) {
          return;
        } else {
          const theNum = Object.keys(response.val()).length;
            this.setState({
              num: theNum,
            })
        }
      })
  }

  render() {
    return (
      <Link to={REVIEW} style={{ textDecoration: 'none' }}>
        <div className="reviewWidget">
          <div className="counterDisplay">{this.state.num}</div>
          <h1>Questions to be reviewed</h1>
        </div>
      </Link>
    )
  }
}

export default QuestionsToReview;
