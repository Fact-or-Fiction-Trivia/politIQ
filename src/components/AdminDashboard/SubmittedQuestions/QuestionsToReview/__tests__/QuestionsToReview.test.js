import React from 'react';
import { shallow } from 'enzyme';

import QuestionsToReview from '../index';

describe('<QuestionsToReview/>', () => {
    it('should render correctly and match snapshot', () => {
        const wrapper = shallow(<QuestionsToReview />);

        expect(wrapper).toMatchSnapshot()
    });

    it('should display the correct number of submitted questions', () => {
        const wrapper = shallow(<QuestionsToReview />);
        // setting the state, like in component did mount
        wrapper.setState({ num: 2 });
        wrapper.update();

        const counterDisplay = wrapper.find('div.counterDisplay');
        expect(parseInt(counterDisplay.text())).toEqual(2);
    });
});