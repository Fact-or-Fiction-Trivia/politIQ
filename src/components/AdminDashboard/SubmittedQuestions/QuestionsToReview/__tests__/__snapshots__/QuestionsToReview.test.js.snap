// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionsToReview/> should render correctly and match snapshot 1`] = `
<Link
  replace={false}
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/review"
>
  <div
    className="reviewWidget"
  >
    <div
      className="counterDisplay"
    >
      0
    </div>
    <h1>
      Questions to be reviewed
    </h1>
  </div>
</Link>
`;
